backend/models/job.js

```diff
@@ -18,155 +18,205 @@
-	return this.findOne({account: teamspace}, {_id: job});
+Job.findUsersWithJobs = async function(teamspace, jobNames) {
+	const jobsColl = await getCollection(teamspace);
+	const foundJobs = await (await jobsColl.find({ _id: { $in: jobNames } })).toArray();
 
+	return foundJobs.reduce((users, jobItem) => users.concat(jobItem.users), []);
 };
 
-schema.statics.findByUser = function(teamspace, user) {
-	return this.findOne({account: teamspace}, {users: user});
+Job.getAllColors = async function(teamspace) {
+	const jobs = await Job.getAllJobs(teamspace);
+	return compact(uniq(map(jobs, "color")));
 };
 
-schema.statics.addUserToJob = function(teamspace, user, jobName) {
-	// Check if user is member of teamspace
-	const User = require("./user");
-	return User.teamspaceMemberCheck(teamspace, user).then(() => {
-		return Job.findByJob(teamspace, jobName).then((job) => {
-			if(!job) {
-				return Promise.reject(responseCodes.JOB_NOT_FOUND);
-			}
-
-			return Job.removeUserFromAnyJob(teamspace, user).then(() => {
-				job.users.push(user);
-				return job.save();
-			});
+Job.getAllJobs = async function(teamspace) {
+	const jobsColl = await getCollection(teamspace);
+	const foundJobs = await (await jobsColl.find()).toArray();
 
-		});
+	return foundJobs.map(({_id, color}) => {
+		return {_id, color};
 	});
 };
 
-schema.statics.addJob = function(teamspace, jobData) {
-	if(!jobData._id || !validateJobName(jobData._id)) {
-		return Promise.reject(responseCodes.JOB_ID_INVALID);
-	}
-	return this.findByJob(teamspace, jobData._id).then(jobFound => {
-		if(jobFound) {
-			return Promise.reject(responseCodes.DUP_JOB);
-		}
+Job.getUserJob = async function(teamspace, user) {
+	const foundJob = await Job.findByUser(teamspace, user);
 
-		const newJobEntry = this.model("Job").createInstance({account: teamspace});
-		newJobEntry._id = jobData._id;
-		if(jobData.color) {
-			newJobEntry.color = jobData.color;
-		}
-		return newJobEntry.save();
-	});
+	return foundJob ? {
+		_id: foundJob._id,
+		color: foundJob.color
+	} : {};
 };
 
-schema.methods.updateJob = function(updatedData) {
-	if(updatedData.color) {
-		this.color = updatedData.color;
+Job.removeJob = async function(teamspace, jobName) {
+	const foundJob = await Job.findByJob(teamspace, jobName);
+
+	if (!foundJob) {
+		throw responseCodes.JOB_NOT_FOUND;
+	}
+
+	if (foundJob.users.length > 0) {
+		throw responseCodes.JOB_ASSIGNED;
 	}
 
-	return this.save();
+	const jobsColl = await getCollection(teamspace);
+	return jobsColl.remove({_id: jobName});
 };
 
-schema.statics.removeJob = function(teamspace, jobName) {
+Job.removeUserFromAnyJob = async function(teamspace, user) {
+	const job = await Job.findByUser(teamspace, user);
+	let result;
 
-	return this.findByJob(teamspace, jobName).then(jobFound => {
-		if(!jobFound) {
-			return Promise.reject(responseCodes.JOB_NOT_FOUND);
-		}
+	if (job) {
+		result = Job.removeUserFromJob(teamspace, job._id, user);
```

Comment:
This function returns a promise which we're not awaiting here?

The result also doesn't seem important considering it's an update?